src/main/java/com/endava/soa_5ed/task4/collections/User.java

```diff
@@ -0,0 +1,59 @@
+package com.endava.soa_5ed.task4.collections;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class User {
+    protected final Map<String, Address> addresses;
+    private long id;
+    private String username;
```

Comment:
You don't have any setter for this fields - so you may make them final too.

Reply:
Done